server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java

```diff
@@ -1066,7 +1070,7 @@ public void test_add_column_not_null_constraint_added_to_a_nested_column() {
 
         // existing object
         execute("alter table t add column o['a']['b'] text not null");
-        Asserts.assertSQLError(() -> execute("insert into t (o) values ({\"a\" = {\"b\" = null}})"))
+        Asserts.assertSQLError(() -> execute("insert into t (o2, o) values (?, {\"a\" = {\"b\" = null}})", new Object[] { ab }))
             .hasPGError(INTERNAL_ERROR)
             .hasHTTPError(BAD_REQUEST, 4000)
             .hasMessageContaining("\"o['a']['b']\" must not be null");
```

Comment:
Due to the new `notNullColumns` ordering, this failed, mentioning that `o2['a']['b']` must not be null - which was also a correct error, but not the point of the test.